src/main/java/io/lettuce/core/RedisClient.java

```diff
@@ -805,15 +805,15 @@ private static void checkValidRedisURI(RedisURI redisURI) {
         }
     }
 
-    private static <K, V> void assertNotNull(RedisCodec<K, V> codec) {
+    protected static <K, V> void assertNotNull(RedisCodec<K, V> codec) {
```

Comment:
A little confused. Seems `LettuceAssert` already solved what you need.

Reply:
The `assertNotNull` methods are not super critical. I can take them out of the PR if needed.

Reply:
Please revert that change and use `LettuceAssert.notNull(…)` instead.